### PR TITLE
Replace hack with SUBSTATION collection impact consumption

### PR DESCRIPTION
**Pull request by sBouzols**

### CODE REVIEW

File: src/components/study-container.js [hunk 1]
```diff
@@ -680,25 +682,28 @@ export function StudyContainer({ view, onChangeTab }) {
...
-                    });
+                const updatedEquipments = fetchAllEquipments(
+                    studyUuid,
+                    currentNode?.id,
+                    collectionElementImpacts?.includes(
+                        EQUIPMENT_TYPES.SUBSTATION
+                    )
+                        ? undefined
+                        : substationsIds
+                );
```
Comment (ne0ds) [inline src/components/study-container.js:693]: Not useful to define this function in a constant

File: src/hooks/use-update-equipments.ts [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+import { UUID } from 'crypto';
+
+function parseStudyNotification(studyUpdatedForce: StudyUpdated) {
+    const payload = studyUpdatedForce.eventData.payload;
+    const substationsIds = payload?.impactedSubstationsIds;
+    const deletedEquipments = payload?.deletedEquipments;
+    const collectionElementImpacts = payload?.collectionElementImpacts;
+
+    return [substationsIds, deletedEquipments, collectionElementImpacts];
+}
```
Comment (ne0ds) [inline src/hooks/use-update-equipments.ts:29]: This function is not very useful.

File: src/hooks/use-update-equipments.ts [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+    const studyUpdatedForce = useSelector(
+        (state: ReduxState) => state.studyUpdated
+    );
+
+    useEffect(() => {
+        if (studyUpdatedForce.eventData.headers) {
+            if (
+                studyUpdatedForce.eventData.headers[UPDATE_TYPE_HEADER] ===
+                'study'
+            ) {
```
Comment (ne0ds) [inline src/hooks/use-update-equipments.ts:56]: The two if statements can be combined

File: src/hooks/use-update-equipments.ts [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+function parseStudyNotification(studyUpdatedForce: StudyUpdated) {
+    const payload = studyUpdatedForce.eventData.payload;
+    const substationsIds = payload?.impactedSubstationsIds;
+    const deletedEquipments = payload?.deletedEquipments;
+    const collectionElementImpacts = payload?.collectionElementImpacts;
+
+    return [substationsIds, deletedEquipments, collectionElementImpacts];
+}
+
+export type FetchEquipmentsPropsType = {
```
Comment (ne0ds) [inline src/hooks/use-update-equipments.ts:31]:
```suggestion
type FetchEquipmentsPropsType = {
```

File: src/hooks/use-update-equipments.ts [hunk 5]
```diff
@@ -0,0 +1,127 @@
...
+    studyUuid: UUID;
+    currentNodeUuid: UUID;
+};
+
+export type DeletedEquipmentType = {
```
Comment (ne0ds) [inline src/hooks/use-update-equipments.ts:36]:
```suggestion
type DeletedEquipmentType = {
```

File: src/redux/reducer.type.ts [hunk 6]
```diff
@@ -42,16 +47,45 @@ export interface StudyUpdatedEventDataHeader {
...
+};
+
+type StudyUpdatedUndefined = {
+    type: undefined;
+    eventData: StudyUpdatedEventDataUnknown;
+};
+
+// Redux state
+export type StudyUpdated = {
+    force: 0 | 1;
```
Comment (ne0ds) [inline src/redux/reducer.type.ts:86]: Why a number?

**Comment by SlimaneAmar** [comment]
REname `collectionElementImpacts` by `impactedElementTypes`

File: src/redux/reducer.type.ts (hunk 6 above)
Comment (sBouzols) [reply src/redux/reducer.type.ts:86]: It's a code which existed before this PR

**Review by SlimaneAmar — APPROVED** [review]

**Review by ne0ds — APPROVED** [review]
Code OK, tests non réalisés.

**Pull request merged**